src/main/java/com/metamx/metrics/KeyedDiff.java

```diff
@@ -41,7 +41,7 @@
 
   public static Map<String, Long> subtract(Map<String, Long> xs, Map<String, Long> ys)
   {
-    assert xs.keySet() == ys.keySet();
+    assert xs.keySet().equals(ys.keySet());
```

Comment:
I don't know how this used to work before, because separately constructed `ImmutableMap.of()` instances are subtracted there, how their keySets could be the same object??

Anyway, this stopped working after I made changes in JvmMonitor, sets are equal but not identical.

Performance is irrelevant because 1) sets are small (4 elements in the case) and metrics are emitted rarely at CPU scale. I've seen CPU profiles from several services and this metrics emitting service takes absolutely insignificant portion of time in them.